include/Reader/readerir.h

```diff
@@ -1605,9 +1618,12 @@ class GenIR : public ReaderBase {
   uint32_t TargetPointerSizeInBits;
   const uint32_t UnmanagedAddressSpace = 0;
   const uint32_t ManagedAddressSpace = 1;
-  llvm::PointerType *ArrayOfReferenceType; ///< Array type for use in casting
-                                           ///< non-array types seen in array
-                                           ///< contexts.
+  llvm::Type *BuiltinObjectType; ///< Cached LLVM representation of object.
+
+  /// \brief Map from Element types to the LLVM representation of the
```

Comment:
When a declaration is short enough to leave room for a comment after it, that seems appropriate. But if a declaration is long I think it is better to have a preceding comment.